scripts/run_driver_tests.psm1

```diff
@@ -67,178 +69,196 @@ function Invoke-Test
         $ArgumentsList += '-s'
     }
 
-    Write-Log "$TestName $ArgumentsList"
-    $Output = &$TestName $ArgumentsList
-    $TestName = $OriginalTestName
-    if ($LASTEXITCODE -ne 0) {
-        throw ("$TestName Failed with $LASTEXITCODE.")
-    }
-
-    # Parse output to check for errors.
-    if ($Output -eq $null) {
-        throw ("No output generated by $TestName.")
-    }
-    Out-String -InputObject $Output | Write-Log
-    $ParsedOutput = $Output.Split(" ")
-    if (-not (($null -ne $ParsedOutput) -and ($ParsedOutput.Length -ge 2))) {
-        throw ("Failed to parse output generated by $TestName.")
-    }
-    if ($ParsedOutput[$ParsedOutput.Length -2] -eq "failed") {
-        throw ("$TestName Test Failed.")
-    }
-    $ParsedOutput = $Output.Split([System.Environment]::NewLine)
-    foreach($line in $ParsedOutput) {
-        if ($line -clike "*FAILED:*") {
-            Write-Host $line
-            throw ("$TestName Test Failed.")
-        }
-    }
+    $JoinedArgumentsList = $ArgumentsList -join " "
+    $TestRunScript = ".\Run-Self-Hosted-Runner-Test.ps1"
+    & $TestRunScript `
+        -TestCommand $TestName `
+        -TestArguments $JoinedArgumentsList `
+        -TestHangTimeout $TestHangTimeout `
+        -UserModeDumpFolder $UserModeDumpFolder `
+        -NeedKernelDump $True `
+        -Verbose
 
     Write-Log "$TestName Passed" -ForegroundColor Green
+    Write-Log "`n`n"
 }
 
 function Invoke-CICDTests
 {
     param([parameter(Mandatory = $true)][bool] $VerboseLogs,
-          [parameter(Mandatory = $false)][bool] $Coverage = $false)
+          [parameter(Mandatory = $false)][bool] $Coverage = $false,
+          [parameter(Mandatory = $false)][int] $TestHangTimeout = 3600,
+          [parameter(Mandatory = $false)][string] $UserModeDumpFolder = "C:\Dumps"
+    )
+
 
     pushd $WorkingDirectory
     $env:EBPF_ENABLE_WER_REPORT = "yes"
 
-    try {
```

Comment:
why is the try0catch no longer necessary?

Reply:
The new test runner script throws exceptions on errors and _not_ catching those ensures that the test is marked as 'failed'.  If there's another way of ensuring that failing tests are marked as 'failed' with the try in place, I'd be happy to consider it.  I'm still ramping up on my PS-fu so any pointers would be definitely appreciated.